lib/control/tests/test_control_cmds.c

```diff
@@ -22,46 +22,62 @@
  *
  */
 
+
 #include "testutils.h"
+#include "lib/messages.h"
+#include "control/control.h"
+#include "lib/control/control-commands.h"
+#include "stats/stats-control.h"
 #include "control/control-server.h"
-#include "control/control-commands.c"
-#include "stats/stats-control.c"
 #include "stats/stats-cluster.h"
 #include "stats/stats-registry.h"
 #include "apphook.h"
 
+static GList *command_list = NULL;
+
+static ControlCommand *
+command_test_get(const char *cmd)
+{
+  GList *command = g_list_find_custom(command_list, cmd, (GCompareFunc)control_command_is_start_with_command);
+  if (NULL == command)
+    return NULL;
+  return (ControlCommand *)command->data;
+}
+
 void
 test_log(void)
 {
   GString *command = g_string_sized_new(128);
   GString *reply;
 
+  CommandFunction control_connection_message_log = command_test_get("LOG")->func;
+
   g_string_assign(command,"LOG");
-  reply = control_connection_message_log(command, NULL);
+  reply = (*control_connection_message_log)(command, NULL);
```

Comment:
I'd remove all explicit function dereference operators from this test, it's redundant.